File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        if(n == 0) return true;
        
        if(flowerbed.size() == 1)
        {
            if(flowerbed[0] != 0 && n > 0) return false;
            if(flowerbed[0] == 0 && n == 1) return true;
            if(flowerbed[0] == 0 && n > 1) return false;
            
        }
        for(int i = 0; i < flowerbed.size(); i++)
        {
            if(i == 0 && flowerbed[i] == 0 && flowerbed[i + 1] == 0)
            {
                n--;
                flowerbed[i] = 1;
            }
            else if(i > 0 && i < flowerbed.size() - 1 && flowerbed[i] == 0 &&  flowerbed[i + 1] == 0 && flowerbed[i - 1] == 0)
            {
                n--;
                flowerbed[i] = 1;
            }
            else if(i == flowerbed.size() - 1 && flowerbed[i] == 0 && flowerbed[i - 1] == 0)
            {
                n--;
                flowerbed[i] = 1;
            }
        }
        
        return n <= 0;
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if n == 0:
            return True
        
        if len(flowerbed) == 1:
            if flowerbed[0] == 1 and n > 0:
                return False
            elif flowerbed[0] == 0 and n == 1:
                return True
            elif flowerbed[0] == 0 and n > 1:
                return False
        
        for i in range(len(flowerbed)):
            if i == 0 and flowerbed[i] == 0 and flowerbed[i + 1] == 0:
                n -= 1
                flowerbed[i] = 1
            elif i == len(flowerbed) - 1 and flowerbed[i] == 0 and flowerbed[i - 1] == 0:
                n -= 1
                flowerbed[i] = 1
            elif i > 0 and i < len(flowerbed) - 1 and flowerbed[i] == 0 and flowerbed[i + 1] == 0 and flowerbed[i - 1] == 0:
                n -= 1
                flowerbed[i] = 1
        
        
        return n <= 0
                
                